sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/extensions/ResultExtensions.kt

```diff
@@ -74,9 +75,15 @@ fun org.researchstack.backbone.result.TaskResult.clientDataAnswerMap(): Map< Str
 fun TaskResult.clientDataAnswerMap(): Map<String, Any> {
     val answerMap = TreeMap<String, Any>()
     this.flattenResult().forEach { result ->
-        (result as? AnswerResult<*>)?.let { answerResult ->
-            answerResult.answer?.let {
-                answerMap.put(answerResult.identifier, it)
+        (result as? ReportResultDataMap)?.let { customDataMap ->
+            customDataMap.toDataMap()?.forEach {
+                answerMap[it.key] = it.value
+            }
+        } ?: run {
```

Comment:
Could this have been used to create the answer.json for archiving instead of ResultListArchiveFactory?

Reply:
Good point!  I updated the archive factory branch to use this instead.  It also fixed the nested TaskResult bug by including only the top-level TaskResult answers map, which will traverse the nested TaskResults and include their answer maps as well.